rcl/include/rcl/guard_condition.h

```diff
@@ -48,6 +49,14 @@ typedef struct rcl_guard_condition_options_s
   rcl_allocator_t allocator;
 } rcl_guard_condition_options_t;
 
+/// On trigger callback data
+typedef struct rcl_guard_condition_callback_data_s
+{
+  rcl_event_callback_t on_trigger_callback;
+  const void * user_data;
+  size_t trigger_count;
```

Comment:
@mauropasse implementation looks good, but could you explain the use case of `trigger_count`?